src/main/java/org/isf/lab/service/LabIoOperations.java

```diff
@@ -103,6 +104,33 @@ public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, Local
 				                                                      TimeTools.truncateToSeconds(dateTo.with(LocalTime.MAX)));
 	}
 	
+	/**
+	 * Return a list of exams ({@link Laboratory}s) between specified dates and matching passed exam name
+	 * @param exam - the exam name as <code>String</code>
+	 * @param dateFrom - the lower date for the range
+	 * @param dateTo - the highest date for the range
+	 * @param patient - the object of patient 
+	 * @return the list of {@link Laboratory}s 
+	 * @throws OHServiceException
+	 */
+	public List<Laboratory> getLaboratory(String exam, LocalDateTime dateFrom, LocalDateTime dateTo, Patient patient) throws OHServiceException {
+		List<Laboratory> laboritories = new ArrayList<>();
+		
+		if (exam.equals("") && patient != null) {
+			laboritories = repository.findByLabDateBetweenAndExamDescriptionAndPatientCode(dateFrom, dateTo, exam, patient.getCode());
+		}
+		if (exam.equals("") && patient == null ) {
+			laboritories = repository.findByLabDateBetweenAndExam_DescriptionOrderByLabDateDesc(dateFrom, dateTo, exam);
+		}
+		if (patient != null && exam.equals("")) {
+			laboritories = repository.findByLabDateBetweenAndPatientCode(dateFrom, dateTo, patient.getCode());
+		}
+		if (patient == null && exam.equals("")) {
+			laboritories= repository.findByCreatedDateBetweenOrderByLabDateDesc(dateFrom, dateTo);
+		}
```

Comment:
@ArnaudFonzam  what is the difference between lines 119 and 125, lines 122 and 128 ?

Reply:
the two first is !exam.equals("")

Reply:
like this?

```
		if (!exam.equals("") && patient != null) {
			laboritories = repository.findByLabDateBetweenAndExamDescriptionAndPatientCode(dateFrom, dateTo, exam, patient.getCode());
		}
		if (!exam.equals("") && patient == null ) {
			laboritories = repository.findByLabDateBetweenAndExam_DescriptionOrderByLabDateDesc(dateFrom, dateTo, exam);
		}
		if (patient != null && exam.equals("")) {
			laboritories = repository.findByLabDateBetweenAndPatientCode(dateFrom, dateTo, patient.getCode());
		}
		if (patient == null && exam.equals("")) {
			laboritories= repository.findByCreatedDateBetweenOrderByLabDateDesc(dateFrom, dateTo);
		}
```

Reply:
yes

Reply:
Ok, I will fix that